### ISSUE TITLE
[contracts] Audit tasks

### ISSUE THREAD

**Issue by rhlsthrm** [ConnextProject/indra#601]
See canonical doc for full description.

**Comment by bohendo** [comment]
Note: blocked by issue #648

**Comment by bohendo** [comment]
This work is effectively done.
Contracts might need further upgrades, we should track those in a separate issue.
These upgrades can't be merged until issue #826 is resolved, closing this issue in favor of that one.
